cvmfs/export_plugin/posix/garbage_collector.cc

```diff
@@ -0,0 +1,204 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+#include "garbage_collector.h"
+
+#include <dirent.h>
+#include <errno.h>
+#include <stdio.h>
+
+#include <map>
+#include <string>
+
+#include "export_plugin/fs_traversal.h"
+#include "helpers.h"
+#include "logging.h"
+#include "smalloc.h"
+#include "util/posix.h"
+
+/**
+ * Initializes garbage collection by setting up the corresponding hash map
+ * in the fs_traversal_posix_context
+ * The initialization uses the data stored in the file at
+ * [data directory] + POSIX_GARBAGE_DIR + POSIX_GARBAGE_FLAGGED_FILE
+ * 
+ * This file contains a fixed length concatenation of
+ * inodes (type ino_t) which should be checked
+ * and deleted upon garbage collection
+ */
+void InitializeGarbageCollection(struct fs_traversal_context *ctx) {
+  struct fs_traversal_posix_context *posix_ctx
+    = reinterpret_cast<struct fs_traversal_posix_context *>(ctx->ctx);
+  int res2 = mkdir((std::string(ctx->data)+POSIX_GARBAGE_DIR).c_str(), 0700);
+    assert(res2 == 0 || errno == EEXIST);
+  if (FileExists(std::string(ctx->data)
+    + POSIX_GARBAGE_DIR + POSIX_GARBAGE_FLAGGED_FILE)) {
```

Comment:
For clarity we may want to compose this name once and then use it. I know it is only twice, then the lines could also be slightly more readable. Not strictly needed, but possibly clearer.